extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisioner.java

```diff
@@ -0,0 +1,177 @@
+    public static final GcpServiceAccount ADC_SERVICE_ACCOUNT = new GcpServiceAccount("adc-email", "adc-name", "application default");
+    private final GcpConfiguration gcpConfiguration;
+    private final BigQueryFactory bqFactory;
+    private final IamService iamService;
+    private final Monitor monitor;
+
+    private BigQueryProvisioner(GcpConfiguration gcpConfiguration, BigQueryFactory bqFactory, IamService iamService, Monitor monitor) {
+        this.gcpConfiguration = gcpConfiguration;
+        this.bqFactory = bqFactory;
+        this.iamService = iamService;
+        this.monitor = monitor;
+    }
+
+    @Override
+    public boolean canProvision(ResourceDefinition resourceDefinition) {
+        return resourceDefinition instanceof BigQueryResourceDefinition;
+    }
+
+    @Override
+    public boolean canDeprovision(ProvisionedResource resourceDefinition) {
+        return resourceDefinition instanceof BigQueryProvisionedResource;
+    }
+
+    @Override
+    public CompletableFuture<StatusResult<ProvisionResponse>> provision(
+            BigQueryResourceDefinition resourceDefinition, Policy policy) {
+        var target = getTarget(resourceDefinition);
+        monitor.info("BigQuery Provisioner provision " + target.getTableName());
+        // TODO check if injected IAM service can be used (ADC / refresher).
+        var serviceAccountName = getServiceAccountName(resourceDefinition);
+
+        var tableName = Optional.ofNullable(target.table())
+                .orElseGet(() -> {
+                    var generatedTableName = resourceDefinition.getId();
+                    monitor.debug("BigQuery Provisioner table name generated: " + generatedTableName);
+                    return generatedTableName;
+                });
+        var resourceName = tableName + "-table";
+
+        // TODO update target with the generated table name.
+
+        try {
+            GcpServiceAccount serviceAccount = null;
+            if (serviceAccountName != null) {
+                serviceAccount = iamService.getServiceAccount(serviceAccountName);
+            }
+
+            var bigQuery = bqFactory.createBigQuery(serviceAccountName);
+            var table = bigQuery.getTable(target.getTableId());
+            if (table == null || !table.exists()) {
+                monitor.warning("BigQuery Provisioner table " + target.getTableName() + " DOESN'T exist");
+                return completedFuture(StatusResult.failure(ResponseStatus.FATAL_ERROR, "Table " + target.getTableName().toString() + " doesn't exist"));
+            }
+            monitor.info("BigQuery Provisioner table " + target.getTableName().toString() + " exists");
+
+            GcpAccessToken token = null;
+
+            if (serviceAccount != null) {
+                token = iamService.createAccessToken(serviceAccount);
+            } else {
+                serviceAccount = ADC_SERVICE_ACCOUNT;
+                token = iamService.createDefaultAccessToken();
+            }
```

Comment:
this piece of code is duplicated in the `GcsProvisioner`, could be the `iamService.createAccessToken` that detects if the service account passed is a default one and in that case it will return the default access token?
This will simplify things here and give the knowledge about accounts to the iam service (that's the one that should have them)